remote_attestation/rust/src/crypto.rs

```diff
@@ -200,13 +206,34 @@ impl KeyNegotiator {
         self,
         peer_public_key: &[u8; KEY_AGREEMENT_ALGORITHM_KEY_LENGTH],
     ) -> anyhow::Result<AeadEncryptor> {
+        let (encryption_key, decryption_key) = self
+            .derive_session_keys(peer_public_key)
+            .context("Couldn't derive session keys")?;
+        let encryptor = AeadEncryptor::new(encryption_key, decryption_key);
+        Ok(encryptor)
+    }
+
+    /// Implementation of the session keys derivation.
+    /// Returns a tuple with an encryption key and a decryption key.
+    pub(crate) fn derive_session_keys(
+        self,
+        peer_public_key: &[u8; KEY_AGREEMENT_ALGORITHM_KEY_LENGTH],
+    ) -> anyhow::Result<(
+        [u8; AEAD_ALGORITHM_KEY_LENGTH],
+        [u8; AEAD_ALGORITHM_KEY_LENGTH],
```

Comment:
These seem way to easy to confuse, could you create a struct?

Reply:
Done